src/main/java/org/rumbledb/items/YearMonthDurationItem.java

```diff
@@ -58,26 +57,26 @@ public boolean isTypeOf(ItemType type) {
 
     @Override
     public void read(Kryo kryo, Input input) {
-        this.value = getDurationFromString(input.readString(), AtomicTypes.YearMonthDurationItem).normalizedStandard(
+        this.value = getDurationFromString(input.readString(), ItemTypes.YearMonthDurationItem).normalizedStandard(
             yearMonthPeriodType
         );
         this.isNegative = this.value.toString().contains("-");
     }
 
     @Override
-    public boolean isCastableAs(AtomicTypes itemType) {
-        return itemType.equals(AtomicTypes.YearMonthDurationItem)
+    public boolean isCastableAs(ItemType itemType) {
+        return itemType.getType() == ItemTypes.YearMonthDurationItem
             ||
-            itemType.equals(AtomicTypes.DayTimeDurationItem)
+            itemType.getType() == ItemTypes.DayTimeDurationItem
             ||
-            itemType.equals(AtomicTypes.DurationItem)
+            itemType.getType() == ItemTypes.DurationItem
             ||
-            itemType.equals(AtomicTypes.StringItem);
+            itemType.getType() == ItemTypes.StringItem;
```

Comment:
Is this intentional to push performance to its limits through bypassing the equals method implementation on ItemType?

Otherwise, I would suggest simply calling `itemType.equals(ItemType.stringItem)`. This facilitates the package private access I mentioned in my other comment by hiding the ItemTypes class from direct access outside ItemType class.

Reply:
Absolutely -- this will all go away with a comparison on the ItemType objects, in a future PR. Then, it will be an equals() call.